packages/jsii-pacmak/lib/targets/golang/types/class.ts

```diff
@@ -1,29 +1,78 @@
-import { Method, ClassType } from 'jsii-reflect';
+import { Method, ClassType, Initializer } from 'jsii-reflect';
 import { CodeMaker } from 'codemaker';
 import { GoTypeRef } from './go-type-reference';
 import { GoStruct } from './go-type';
 import { TypeField } from './type-field';
+import { getFieldDependencies, substituteReservedWords } from '../util';
 import { Package } from '../package';
+import { ClassConstructor, MethodCall } from '../runtime';
+
+export class GoClassConstructor {
+  private readonly constructorRuntimeCall: ClassConstructor;
+
+  public constructor(
+    public readonly parent: GoClass,
+    private readonly type: Initializer,
+  ) {
+    this.constructorRuntimeCall = new ClassConstructor(this);
+  }
+
+  public emit(code: CodeMaker) {
+    const constr = `New${code.toPascalCase(this.parent.name)}`;
+    const params = this.type.parameters.map(
+      (x) =>
+        `${substituteReservedWords(x.name)} ${new GoTypeRef(
+          this.parent.parent.root,
+          x.type,
+        ).scopedName(this.parent.parent)}`,
+    );
+
+    const parameters = params.length === 0 ? '' : params.join(', ');
+
+    let docstring = '';
+    if (this.type.docs.summary) {
+      docstring = this.type.docs.toString();
+      code.line(`// ${docstring}`);
+    }
+
+    code.openBlock(
+      `func ${constr}(${parameters}) ${code.toPascalCase(
+        this.parent.interfaceName,
```

Comment:
Is `parent.interfaceName` not already in camelCase? It should have been formatted in the construction of the go-type: https://github.com/aws/jsii/blob/master/packages/jsii-pacmak/lib/targets/golang/types/go-type.ts#L115

Reply:
It wasn't. I removed all inline calls to `toPascalCase` in emit and made `GoType` and other classes set their name properties with the correct formatting.